freshmaker/pyxis.py

```diff
@@ -0,0 +1,246 @@
+            bundles_per_index_image[path] = self._pagination('operators/bundles',
+                                                             request_params)
+
+        return bundles_per_index_image
+
+    def get_latest_bundles(self, index_images, params={}):
+        """
+        Get latest bundle images per channel per index image
+
+        :param list index_images: list of index images
+        :param dict params: parameters to add to GET request
+        :return: latest bundle images per channel per index image
+        :rtype: list
+        """
+        bundles_per_index_image = \
+            self._get_bundles_per_index_image(index_images, params)
+
+        ret_bundles = []
+        for index_image, bundles in bundles_per_index_image.items():
+            bundle_per_channel = {}
+            # get latest versions of bundle images per channel
+            for bundle in bundles:
+                channel = bundle['channel_name']
+                try:
+                    if channel in bundle_per_channel:
+                        new_ver = bundle['version']
+                        old_ver = bundle_per_channel[channel]['version']
+                        if version.Version(new_ver) > version.Version(old_ver):
+                            bundle_per_channel[channel] = bundle
+                    else:
+                        # Check if the right format of version is used
+                        version.Version(bundle.get('version'))
+                        bundle_per_channel[channel] = bundle
+                # Check if the right format of version is used
+                except version.InvalidVersion as e:
+                    path = bundle.get('bundle_path', 'Unknown bundle path')
+                    log.warning("Other format than SemVer is used in "
+                                "bundle: %s", path)
+                    log.warning(repr(e))
+            ret_bundles.extend(bundle_per_channel.values())
+
+        return ret_bundles
+
+    def get_digests_by_nvrs(self, nvrs, params=None):
+        """
+        Get images' digests(manifest_list_digest field) by their nvrs
+
+        :param list nvrs: list of nvrs of ContainerImages to query Pyxis
+        :param dict params: parameters to add to GET request
+        :return: set of digests
+        :rtype: set
+        """
+        if params is None:
+            request_params = {}
+        else:
+            request_params = copy.deepcopy(params)
+        request_params['include'] = ','.join([
+            'data.brew', 'data.repositories'])
+
+        # get all manifest_list_digest of ContainerImages we got from Pyxis
+        digests = set()
+        for nvr in nvrs:
+            for image in self._pagination(f'images/nvr/{nvr}', request_params):
+                for repo in image.get('repositories'):
+                    if repo['published'] and 'manifest_list_digest' in repo:
+                        digests.add(repo['manifest_list_digest'])
```

Comment:
@apaplaus I was incorrect in my last statement. The manifest list digest should not change based on the registry that it is on. Still, let's be safe and only used the published repo but we can break after finding the first one.
```suggestion
                        digests.add(repo['manifest_list_digest'])
                        break
```